monasca_events_api/tests/first_test.py

```diff
@@ -12,10 +12,246 @@
+from monasca_events_api.common.messaging import exceptions
+from monasca_events_api.common.repositories.mysql.events_repository import EventsRepository
+from monasca_events_api.v2.events import Events
+
+import mock
+from monasca_events_api.common.repositories.exceptions import RepositoryException
+
+from oslo_utils import timeutils
+
 import unittest
 
 
+class EventsSubClass(Events):
+    def __init__(self):
+        self._default_authorized_roles = ['user', 'domainuser',
+                                          'domainadmin', 'monasca-user']
+        self._post_events_authorized_roles = ['user', 'domainuser',
+                                              'domainadmin', 'monasca-user', 'monasca-agent']
+        self._events_repo = None
+        self._message_queue = None
+        self._region = 'useast'
+
+    def _event_transform(self, event, tenant_id, _region):
+        return dict(event=1, meta=dict(tenantId='0ab1ac0a-2867-402d',
+                    region='useast'), creation_time=timeutils.utcnow_ts())
+
+
 class Test_first(unittest.TestCase):
 
-    def test_first(self):
-        assert 1 == 1
+    @mock.patch('monascaclient.ksclient.KSClient')
+    def _generate_req(self, token):
+        """Generate a mock HTTP request"""
+        req = mock.MagicMock()
+        req.get_param.return_value = None
+
+        req.headers = {
+            'X-Auth-User': 'mini-mon',
+            'X-Auth-Token': token,
+            'X-Auth-Key': 'password',
+            'X-TENANT-ID': '0ab1ac0a-2867-402d',
+            'X-ROLES': 'user, domainuser, domainadmin, monasca-user, monasca-agent',
+            'Accept': 'application/json',
+            'User-Agent': 'python-monascaclient',
+            'Content-Type': 'application/json'}
+
+        req.body = {}
+        req.content_type = 'application/json'
+        return req
+
+    @mock.patch('monasca_events_api.v2.events.Events._list_event')
+    @mock.patch('monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_get_pass_singleevent(self, helper_tenant_id, helpers_validate, mysqlRepo, listev):
+        """GET Method success Single Event"""
+        helpers_validate.validate_authorization.return_value = True
+        returnEvent = [{"region": "useast", "tenantId": "0ab1ac0a-2867-402d",
+                        "creation_time": "1434331190", "event": "1"}]
+        listev.return_value = returnEvent
+
+        mysqlRepo.connect.return_value = True
+        helper_tenant_id.get_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        event_id = "1"
+        eventsObj = EventsSubClass()
+        eventsObj._events_repo = EventsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        eventsObj.on_get(self._generate_req(), res, event_id)
+        self.assertEqual(returnEvent, eval(res.body))
```

Comment:
Rather than eval you can use json.loads here to convert the json body back into a python dict.  It's safer than eval and makes it more clear what the data actually is.